Add marker clusters to the k5 pd map, clean up some related code

aoby opened the pull request:
… and remove 2 warnings

Now, multiple close pins on the [K5 workshop map](https://code.org/professional-development-workshops) will be [clustered](https://developers.google.com/maps/documentation/javascript/marker-clustering#adding-a-marker-clusterer):
![image](https://user-images.githubusercontent.com/8405316/27494074-22c92f8e-5801-11e7-9ccc-bad6bee3b599.png)

Clicking the cluster zooms in to reveal the individual pins:
![image](https://user-images.githubusercontent.com/8405316/27494127-5b53c0b2-5801-11e7-891f-5ec44eea68a8.png)

Note: the number represents the number of pins, meaning unique locations of workshops. Since we already group workshops by location, this number will not represent the total number of workshops where there are more than one in a given location.

----

This PR also fixes 2 warnings from the google map apis, as seen on production:
![image](https://user-images.githubusercontent.com/8405316/27494182-990e0e58-5801-11e7-951e-b95ece2ed864.png)
 1. [SensorNotRequired](https://developers.google.com/maps/documentation/javascript/error-messages#sensor-not-required
) - removed the sensor param.

 2. [RetiredVersion](https://developers.google.com/maps/documentation/javascript/error-messages#retired-version) - we were erroneously requesting version 3.7, but [there is no 3.7](https://developers.google.com/maps/documentation/javascript/versions#documentation-for-the-api-versions). It should have been 3.27. 

CC @tanyaparker

Hamms commented on pegasus/sites.v3/code.org/public/js/workshop-search.js:
```diff
@@ -23,83 +25,89 @@ function initializeMap() {
 function loadWorkshops() {
-  var params = [];
+  markerClusterer = new MarkerClusterer(gmap, [],
+    {imagePath: 'https://developers.google.com/maps/documentation/javascript/examples/markerclusterer/m'}
```
this path seems like a placeholder

aoby replied:
D'oh! Thanks. I apparently didn't read this [part](https://developers.google.com/maps/documentation/javascript/marker-clustering#adding-a-marker-clusterer)

> Get the marker clustering library and images from GitHub, and store them on a server accessible to your app. 

Will update...

aoby replied:
On that note, this might be a good time to move this js over to our js pipeline under `apps` and leverage the [npm module](https://www.npmjs.com/package/js-marker-clusterer).

Hamms commented on pegasus/sites.v3/code.org/public/js/workshop-search.js:
```diff
@@ -23,83 +25,89 @@ function initializeMap() {
+
+      var infoWindowContent = '';
+
+      if (typeof markersByLocation[hash] === 'undefined') {
```
nit: why `typeof` and not just `markersByLocation[hash] === undefined`?

aoby replied:
Good question. I didn't modify [that line](https://github.com/code-dot-org/code-dot-org/blob/staging/pegasus/sites.v3/code.org/public/js/workshop-search.js#L36). I'll update it too...

Hamms commented on pegasus/sites.v3/code.org/views/workshop_search.haml:
```diff
@@ -1,4 +1,5 @@
-%script{type: "text/javascript", src: "https://maps.googleapis.com/maps/api/js?client=#{CDO.google_maps_client_id}&sensor=true&libraries=places,geometry&v=3.7"}
+%script{type: "text/javascript", src: "https://maps.googleapis.com/maps/api/js?client=#{CDO.google_maps_client_id}&libraries=places,geometry&v=3.27"}
+%script{type: "text/javascript", src: "https://developers.google.com/maps/documentation/javascript/examples/markerclusterer/markerclusterer.js"}
```
similarly, this doesn't seem like a valid production url. Should we just add this to the repo?

aoby commented:
I moved `workshop_search.js` into apps, leveraged an NPM package version of the marker clusterer js, and added the cluster image files into our repo to be served as assets.

PTAL
